src/Algebra/Graph.hs

```diff
@@ -423,21 +442,28 @@ overlays = fromMaybe empty . foldr1Safe overlay
 -- 'isEmpty' . connects == 'all' 'isEmpty'
 -- @
 connects :: [Graph a] -> Graph a
-connects = fromMaybe empty . foldr1Safe connect
-{-# INLINE [1] connects #-}
+connects xs = buildg $ \e v o c -> combineR e c (foldg e v o c) xs
+{-# INLINE connects #-}
+
+-- Safe version of foldr with a map (the composition is optimized)
+-- This is a good consumer of lists.
+combineR :: c -> (c -> c -> c) -> (a -> c) -> [a] -> c
+combineR e o f = fromMaybe e . foldr1Safe o . map f
 
 -- | Generalised 'Graph' folding: recursively collapse a 'Graph' by applying
 -- the provided functions to the leaves and internal nodes of the expression.
 -- The order of arguments is: empty, vertex, overlay and connect.
 -- Complexity: /O(s)/ applications of given functions. As an example, the
 -- complexity of 'size' is /O(s)/, since all functions have cost /O(1)/.
 --
+-- Good consumer.
+--
 -- @
--- foldg 'empty' 'vertex'        'overlay' 'connect'        == id
+-- foldg 'empty' 'vertex'        'overlay' 'connect'       == id
```

Comment:
I think the alignment was fine in the Haddock:

https://hackage.haskell.org/package/algebraic-graphs-0.4/docs/Algebra-Graph.html#v:foldg

Are these changes accidental?

Reply:
They are not, the alignment was buggy on master for me. Could you check? Maybe my haddock has a problem.

Reply:
Hmm, that's strange, let me check.